### DISCUSSION TITLE
How to Implement Operators for Sequence Data Calculation?

### DISCUSSION THREAD

**Discussion by Malibu351** [MilesCranmer/PySR#556] [Q&A]
Hello, and thank you for your work. I've been exploring some operators for sequence data recently, such as calculating the mean of sequence data. However, I noticed that PySR operates on individual data pairs (x1, x2, x3, ...) - (y1) rather than matrices or sequences. Could you advise on how I might configure or modify PySR to achieve the functionality I'm looking for? Thank you in advance for your response.

for example:
I want to use `shift(1)`, when I input series (x1, x2, ... , xn), I can get  ( x2, x3, ..., xn, Nan)
or
use `mean(x)`, when I input series (x1, x2, ... , xn) , I can get `avg(x)`, (x1+x2+...+xn)/n

**Comment by MilesCranmer** [comment]
Related:

- https://github.com/MilesCranmer/PySR/discussions/426
- https://github.com/MilesCranmer/PySR/discussions/401
- https://github.com/MilesCranmer/PySR/discussions/275
- https://github.com/MilesCranmer/PySR/discussions/181

**Reply by Malibu351** [reply]
Thans a lot, I read #275 before, and I will read others and study how to fix the problem.
